src/librustc_resolve/late.rs

```diff
@@ -1573,7 +1583,9 @@ impl<'a, 'b> LateResolutionVisitor<'a, '_> {
             let def_id = this.parent_scope.module.normal_ancestor_id;
             let node_id = this.r.definitions.as_local_node_id(def_id).unwrap();
             let better = res.is_some();
-            this.r.use_injections.push(UseError { err, candidates, node_id, better });
+            let suggestion =
+                if res.is_none() { this.report_missing_type_error(path) } else { None };
```

Comment:
Aha, it's now in `smart_resolve_path`, I like it much better.
This is what I though about in https://github.com/rust-lang/rust/pull/68447#discussion_r370348069 basically, except the reporting condition is different.
As long as it's contained inside `diagnostics.rs` I don't care too much about the precise conditions though.